runtime/compiler/aarch64/codegen/J9AheadOfTimeCompile.cpp

```diff
@@ -36,8 +40,1040 @@ void J9::ARM64::AheadOfTimeCompile::processRelocations()
+         *(uintptrj_t*)cursor = (uintptrj_t) aotCI->_constantPool;
+         cursor += SIZEPOINTER;
+
+         *(uintptrj_t*)cursor = (uintptrj_t) aotCI->_cpIndex;
+         cursor += SIZEPOINTER;
+
+         void *romClass = (void *)fej9->getPersistentClassPointerFromClassPointer(aotCI->_clazz);
+         uintptr_t romClassOffsetInSharedCache = sharedCache->offsetInSharedCacheFromPointer(romClass);
+         *(uintptrj_t *)cursor = romClassOffsetInSharedCache;
+         cursor += SIZEPOINTER;
+         }
+         break;
+
+      case TR_ValidateArbitraryClass:
+         {
+         TR::AOTClassInfo *aotCI = (TR::AOTClassInfo*) relocation->getTargetAddress2();
+         TR_OpaqueClassBlock *classToValidate = aotCI->_clazz;
+
+         //store the classchain's offset for the classloader for the class
+         uintptrj_t classChainOffsetInSharedCacheForCL = sharedCache->getClassChainOffsetOfIdentifyingLoaderForClazzInSharedCache(classToValidate);
+         *(uintptrj_t *)cursor = classChainOffsetInSharedCacheForCL;
+         cursor += SIZEPOINTER;
+
+         //store the classchain's offset for the class that needs to be validated in the second run
+         void *romClass = (void *)fej9->getPersistentClassPointerFromClassPointer(classToValidate);
+         uintptrj_t *classChainForClassToValidate = (uintptrj_t *) aotCI->_classChain;
+         uintptr_t classChainOffsetInSharedCache = sharedCache->offsetInSharedCacheFromPointer(classChainForClassToValidate);
+         *(uintptrj_t *)cursor = classChainOffsetInSharedCache;
+         cursor += SIZEPOINTER;
+         }
+         break;
+
+      case TR_HCR:
+         {
+         flags = 0;
+         if (((TR_HCRAssumptionFlags)((uintptrj_t)(relocation->getTargetAddress2()))) == needsFullSizeRuntimeAssumption)
+        	   flags = needsFullSizeRuntimeAssumption;
+         TR_ASSERT((flags & RELOCATION_CROSS_PLATFORM_FLAGS_MASK) == 0,  "reloFlags bits overlap cross-platform flags bits\n");
+         *flagsCursor |= (flags & RELOCATION_RELOC_FLAGS_MASK);
+
+         *(uintptrj_t*) cursor = (uintptrj_t) relocation->getTargetAddress();
+         cursor += SIZEPOINTER;
+         }
+         break;
+
+      case TR_DebugCounter:
+         {
+         TR::DebugCounterBase *counter = (TR::DebugCounterBase *) relocation->getTargetAddress();
+         if (!counter || !counter->getReloData() || !counter->getName())
+            comp->failCompilation<TR::CompilationException>("Failed to generate debug counter relo data");
+
+         TR::DebugCounterReloData *counterReloData = counter->getReloData();
+
+         uintptrj_t offset = (uintptrj_t)fej9->sharedCache()->rememberDebugCounterName(counter->getName());
+
+         uint8_t flags = (uint8_t)counterReloData->_seqKind;
+         TR_ASSERT((flags & RELOCATION_CROSS_PLATFORM_FLAGS_MASK) == 0,  "reloFlags bits overlap cross-platform flags bits\n");
+         *flagsCursor |= (flags & RELOCATION_RELOC_FLAGS_MASK);
```

Comment:
Since according to eclipse/omr#4322 (comment) there is only going to be one materialization sequence, you don't need to set the flag. Additionally, the seq isn't even passed in:
https://github.com/eclipse/omr/blob/0cededa77eb2bc0e54ad27520df748747819272b/compiler/aarch64/codegen/OMRTreeEvaluator.cpp#L215

Reply:
I agree. I removed these 3 lines. I think I forgot to remove these when I copied the code from the P version of J9AheadOfTimeCompile.cpp.